### PR TITLE
Document client connection quotas

### PR DESCRIPTION
**Pull request by shorgi**
Fixes:
- #354 

Preview:
- New topic [Configure Availability](https://deploy-preview-1109--redpanda-documentation.netlify.app/docs/manage/cluster-maintenance/configure-availability/)

**Review deadline**: Fri Jan 27

### CODE REVIEW

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- [kafka_connection_rate_limit](../../../reference/cluster-properties#kafka_connection_rate_limit). This property limits the maximum rate of connections created per second. It applies per CPU core.
+- [kafka_connection_rate_limit_overrides](../../../reference/cluster-properties#kafka_connection_rate_limit_overrides). For targeted overrides of `kafka_connection_rate_limit`, this property enables unlimited connection creation to the IP addresses specified in the property's list.
+
+:::note
+All these connection limit properties are disabled by default, so they require manual configuration to enable.
+:::
+
+:::note
+A client typically opens two or three connections, so the total number of connections usually is not equal to the number of clients.
+:::
```
Comment (micheleRP) [inline docs/manage/cluster-maintenance/configure-availability.mdx:34]:
```suggestion
:::note notes
- These connection limit properties are disabled by default. You must manually enable them.
- Typically, a client opens two or three connections, so the total number of connections is not equal to the number of clients.
:::
```

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import ConfigureAvailabilityIntro from './shared/_intro-configure-availability.mdx'
+
+<ConfigureAvailabilityIntro/>
+
+## Limit Client Connections
+
+A malicious or compromised Kafka client may create many network connections to execute its attacks. To mitigate the risk of a bad client creating too many connections and taking down a node, you can configure a Redpanda cluster to impose limits on the number of created client connections for its Kafka protocol server nodes. 
+
+The cluster properties that limit the number of connections:
```
Comment (micheleRP) [inline docs/manage/cluster-maintenance/configure-availability.mdx:18]:
```suggestion
The following cluster properties limit the number of connections:
```

File: docs/reference/cluster-properties.mdx [hunk 3]
```diff
@@ -318,6 +318,30 @@ Overrides the maximum connections per second for one core for the specified IP a
...
+
+**Restart required**: no
+
+---
+
+### kafka_connections_max_per_ip
+
+Maximum number of Kafka client connections per IP address, per broker. If `null`, the property is disabled.
+
+**Units**: number of Kafka client connections per IP address, per broker.
```
Comment (micheleRP) [inline docs/reference/cluster-properties.mdx:337]:
```suggestion
**Units**: number of Kafka client connections per IP address, per broker
```

File: docs/reference/cluster-properties.mdx [hunk 4]
```diff
@@ -318,6 +318,30 @@ Overrides the maximum connections per second for one core for the specified IP a
 
 ---
 
+### kafka_connections_max
+
+Maximum number of Kafka client connections per broker. If `null`, the property is disabled.
+
+**Units**: number of Kafka client connections per broker.
```
Comment (micheleRP) [inline docs/reference/cluster-properties.mdx:325]:
```suggestion
**Units**: number of Kafka client connections per broker
```

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+- [kafka_connections_max](../../../reference/cluster-properties#kafka_connections_max). Similar to Kafka's `max.connections`, this property sets the maximum number of connections created by a node.
+- [kafka_connections_max_per_ip](../../../reference/cluster-properties#kafka_connections_max_per_ip). Similar to Kafka's `max.connections.per.ip`, this property sets the maximum number of connections created per IP address by a node.
```
Comment (micheleRP) [inline docs/manage/cluster-maintenance/configure-availability.mdx:21]:
```suggestion
- [kafka_connections_max](../../../reference/cluster-properties#kafka_connections_max): Similar to Kafka's `max.connections`, this property sets the maximum number of connections created by a node.
- [kafka_connections_max_per_ip](../../../reference/cluster-properties#kafka_connections_max_per_ip): Similar to Kafka's `max.connections.per.ip`, this property sets the maximum number of connections created per IP address by a node.
```

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+Redpanda also provides properties to manage the rate of connection creation:
+
+- [kafka_connection_rate_limit](../../../reference/cluster-properties#kafka_connection_rate_limit). This property limits the maximum rate of connections created per second. It applies per CPU core.
+- [kafka_connection_rate_limit_overrides](../../../reference/cluster-properties#kafka_connection_rate_limit_overrides). For targeted overrides of `kafka_connection_rate_limit`, this property enables unlimited connection creation to the IP addresses specified in the property's list.
```
Comment (micheleRP) [inline docs/manage/cluster-maintenance/configure-availability.mdx:26]:
```suggestion
- [kafka_connection_rate_limit](../../../reference/cluster-properties#kafka_connection_rate_limit): This property limits the maximum rate of connections created per second. It applies per CPU core.
- [kafka_connection_rate_limit_overrides](../../../reference/cluster-properties#kafka_connection_rate_limit_overrides): For targeted overrides of `kafka_connection_rate_limit`, this property enables unlimited connection creation to the IP addresses specified in the property's list.
```

**Review by micheleRP — APPROVED** [review]
looks good, just minor edits!

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 7]
```diff
@@ -15,20 +15,17 @@ import ConfigureAvailabilityIntro from './shared/_intro-configure-availability.m
...
 
-:::note
-All these connection limit properties are disabled by default, so they require manual configuration to enable.
-:::
-
-:::note
-A client typically opens two or three connections, so the total number of connections usually is not equal to the number of clients.
+:::note notes
+- These connection limit properties are disabled by default. You must manually enable them.
+- Typically, a client opens two or three connections, so the total number of connections is not equal to the number of clients.
```
Comment (jcsp) [inline docs/manage/cluster-maintenance/configure-availability.mdx:30]: To really emphasize this, maybe add "...for example, to support 100 clients, you might set your connection limit to 300".

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+<ConfigureAvailabilityIntro/>
+
+## Limit Client Connections
+
+A malicious Kafka client application may create many network connections to execute its attacks. A poorly configured application may also create an excessive number of connections. To mitigate the risk of a client creating too many connections and using too many system resources, you can configure a Redpanda cluster to impose limits on the number of created client connections for its Kafka protocol server nodes. 
+
+The following cluster properties limit the number of connections:
+
+- [kafka_connections_max](../../../reference/cluster-properties#kafka_connections_max): Similar to Kafka's `max.connections`, this property sets the maximum number of connections created by a node.
+- [kafka_connections_max_per_ip](../../../reference/cluster-properties#kafka_connections_max_per_ip): Similar to Kafka's `max.connections.per.ip`, this property sets the maximum number of connections created per IP address by a node.
```
Comment (jcsp) [inline docs/manage/cluster-maintenance/configure-availability.mdx:21]: There is also `kafka_connections_max_overrides` similar to the other override property for the rate limit, but for connection count.

File: docs/manage/cluster-maintenance/configure-availability.mdx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+Redpanda also provides properties to manage the rate of connection creation:
+
+- [kafka_connection_rate_limit](../../../reference/cluster-properties#kafka_connection_rate_limit): This property limits the maximum rate of connections created per second. It applies per CPU core.
+- [kafka_connection_rate_limit_overrides](../../../reference/cluster-properties#kafka_connection_rate_limit_overrides): For targeted overrides of `kafka_connection_rate_limit`, this property enables unlimited connection creation to the IP addresses specified in the property's list.
```
Comment (jcsp) [inline docs/manage/cluster-maintenance/configure-availability.mdx:26]: It's less that it's unlimited connections, more that you get to specify a limit per IP.

File: docs/reference/cluster-properties.mdx [hunk 10]
```diff
@@ -318,6 +318,30 @@ Overrides the maximum connections per second for one core for the specified IP a
...
+
+**Units**: number of Kafka client connections per broker
+
+**Default**: null
+
+**Restart required**: no
+
+---
+
+### kafka_connections_max_per_ip
```
Comment (jcsp) [inline docs/reference/cluster-properties.mdx:333]: Also kafka_connections_max_overrides, unless that was already in the page

**Review by jcsp — APPROVED** [review]

**Pull request merged**